src/App.js

```diff
@@ -1,11 +1,137 @@
-import React, { useState } from "react";
+import React, { useState, useRef, useEffect, useCallback } from "react";
 import styles from "./App.module.css";
+import "./index.css";
+
+const Tab = ({ children, selected = false, id, index, onClick }) => {
+  const ref = useRef(null);
+  useEffect(() => {
+    if (selected && ref.current) {
+      ref.current.focus();
+    }
+  }, [selected]);
+
+  return (
+    <button
+      role="tab"
+      type="button"
+      aria-selected={selected}
+      aria-controls={`${id}-tab`}
+      id={`${id}-tab`}
+      tabIndex={selected ? 0 : -1}
+      onClick={() => onClick(index)}
+      ref={ref}
+    >
+      {children}
+    </button>
+  );
+};
+
+const TabList = ({ children, label, onKeyDown }) => (
+  <div role="tablist" aria-label={label} onKeyDown={onKeyDown}>
+    {children}
+  </div>
+);
+
+const TabPanel = ({ children, id, selected = false }) => (
+  <div
+    tabIndex={0}
+    role="tabpanel"
+    id={`${id}-tab-panel`}
+    aria-labelledby={id}
+    hidden={!selected}
+  >
+    {children}
+  </div>
+);
+
+const Tabs = ({ content, onDelete }) => {
+  const [activeTab, setActiveTab] = useState(0);
+  const onKeyDown = useCallback(
+    e => {
+      switch (e.key) {
+        case "ArrowRight":
+          e.preventDefault();
+          setActiveTab(activeTab + 1 === content.length ? 0 : activeTab + 1);
+          break;
+        case "ArrowLeft":
+          e.preventDefault();
+          setActiveTab(activeTab === 0 ? content.length - 1 : activeTab - 1);
+          break;
+        case "Home":
+          e.preventDefault();
+          setActiveTab(0);
+          break;
+        case "End":
+          e.preventDefault();
+          setActiveTab(content.length - 1);
+          break;
+        case "Delete":
+          e.preventDefault();
+          setActiveTab(activeTab === 0 ? content.length - 1 : activeTab - 1);
```

Comment:
Why make tab active before deleting it? I wonder where focus will go after we delete element from the DOM 🤔